lib/Models/CatalogGroup.js

```diff
@@ -412,4 +402,135 @@ CatalogGroup.prototype.sortItems = function(sortRecursively) {
     }
 };
 
+/**
+ * Opens this CatalogGroup and all its parents.
+ */
+CatalogGroup.prototype.openWithAllParents = function() {
+    this.isOpen = true;
+
+    if (this.parent) {
+        this.parent.openWithAllParents();
+    }
+};
+
+/**
+ * Does a complete traversal of this CatalogGroup and any CatalogGroups under it and produces a flat map of each item's
+ * share key(s) to that item, using {@link CatalogMember#allShareKeys}. Will throw an Error if there are any duplicate
+ * keys, as this shouldn't ever occur.
+ *
+ * @returns {Object} a map of share keys to items.
+ */
+CatalogGroup.prototype.generateShareKeyIndex = function() {
+    return this.items.reduce(function(indexSoFar, item) {
+        item.allShareKeys.forEach(function(key) {
+            addToIndex(indexSoFar, key, item);
+        });
+
+        if (item.generateShareKeyIndex) {
+            var shareKeyIndex = item.generateShareKeyIndex();
+
+            Object.keys(shareKeyIndex).forEach(function(key) {
+                addToIndex(indexSoFar, key, shareKeyIndex[key]);
+            });
+        }
+
+        return indexSoFar;
+    }, {});
+};
+
+/**
+ * Adds an item to the provided index under the provided key, checking to make sure there's no item for that key already.
+ * Will throw an error if so.
+ */
+function addToIndex(indexSoFar, key, item) {
+    if (indexSoFar[key]) {
+        throw new DeveloperError('Duplicate shareKey: ' + key);
```

Comment:
I think this should be a `RuntimeError`, not a `DeveloperError`, because it's probably caused by a duplicate key in the init file, not by a coding error, right?

As a general rule, we should be able to remove all code that checks for and throws `DeveloperError`, and our app will still work correctly.  In fact, Cesium does exactly this in release builds.